homeassistant/components/refoss_rpc/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""The Refoss RPC integration."""
+
+from __future__ import annotations
+
+from typing import Final
+
+from aiorefoss.common import ConnectionOptions
+from aiorefoss.exceptions import (
+    DeviceConnectionError,
+    InvalidAuthError,
+    MacAddressMismatchError,
+)
+from aiorefoss.rpc_device import RpcDevice
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MAC,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import LOGGER
+from .coordinator import (
+    RefossConfigEntry,
+    RefossCoordinator,
+    RefossEntryData,
+    RefossPollingCoordinator,
+)
+
+PLATFORMS: Final = [
+    Platform.BINARY_SENSOR,
+    Platform.BUTTON,
+    Platform.EVENT,
+    Platform.SENSOR,
+    Platform.SWITCH,
+    Platform.UPDATE,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: RefossConfigEntry) -> bool:
+    """Set up Refoss RPC from a config entry."""
+    if not entry.data.get(CONF_HOST):
+        LOGGER.warning(
+            (
+                "The config entry %s probably comes from a custom integration, please"
+                " remove it if you want to use core Refoss RPC integration"
+            ),
+            entry.title,
+        )
+        return False
```

Comment:
Maybe it is better to create a repair here: https://developers.home-assistant.io/docs/core/platform/repairs/